xds/internal/internal.go

```diff
@@ -34,23 +34,22 @@ type LocalityID struct {
 	SubZone string `json:"subZone,omitempty"`
 }
 
-// String generates a string representation of LocalityID by adding ":" between
-// the components of the LocalityID.
-func (l LocalityID) String() string {
-	return fmt.Sprintf("%s:%s:%s", l.Region, l.Zone, l.SubZone)
+// ToJSON generates a string representation of LocalityID by marshalling it into
+// json.
+func (l LocalityID) ToJSON() (string, error) {
```

Comment:
Looking at the changes elsewhere...should we just delete this function (and its inverse) and call the json package directly instead?  They don't seem to add much.

Reply:
But that hardcodes json in the balancers? Using json should be internal to the locality id.

Maybe I should call this function ToString() and FromString() instead of To/From JSON?

Reply:
I still don't like calling json.Marshall directly. Because it adds the implicit contract that all locality ID string in the balancers are JSON. And if I were to add a new code to convert a string ID into a struct, I need to know that it's JSON, instead of just calling the function that does the job. And to remind people of that, I would need to add comments to all the fields where ID string is stored, and say it is JSOn.

And there's the minor annoyance that Marshal returns []byte that needs to be explicitly converted into string.

I'm keeping these two functions, but calling them ToString and FromString. Let me know what you think.

Reply:
> Because it adds the implicit contract

This is `internal`, so I'm not sure any concerns like this are important.

> Let me know what you think.

This is fine, since it's short-lived.